apps/api/src/app/utils/str-to-slug.ts

```diff
@@ -0,0 +1,136 @@
+import { Not } from "typeorm";
+import * as dotenv from 'dotenv';
+import { join } from "path";
+import { getDataSource } from "./database";
+
+
+dotenv.parse(join(process.cwd(), '.env'));
+
+export function convertToSlug(str) {
+
+  //replace all special characters | symbols with a space
+  str = str.replace(/[`~!@#$%^&*()_\-+=\\[\]{};:'"\\|\\/,.<>?\s]/g, ' ')
+    .toLowerCase();
+
+  // trim spaces at start and end of string
+  str = str.replace(/^\s+|\s+$/gm, '');
+
+  // replace space with dash/hyphen
+  str = str.replace(/\s+/g, '-');
+
+  return str;
+}
+
+export async function generateSlug(type, str: string, exclude?: number | string, slugColumnName = 'slug') {
+  let index = 1;
+  let found: any;
+  let slug: string;
+  do {
+    slug = convertToSlug(str + (index > 1 ? ' ' + index : ''));
+    const dataSource  = getDataSource();
+    found = await dataSource
+      .createQueryBuilder()
+      .select('id')
+      .from(type, 'slug_table')
+      .withDeleted()
+      .where({
+        [slugColumnName]: slug,
+        ...(exclude ? { id: Not(exclude) } : {})
+      })
+      .getRawOne();
+
+    index++;
+  } while (found);
+
+  return slug;
+}
+
+export async function generateRandomUnpickString(type, limit = 6, ColumnName: string) {
```

Comment:
Make this function in new file, file name is `str-to-slug` so keep meaning code in file.